retrier.go

```diff
@@ -275,6 +314,8 @@ func (r *Retrier) sleepOrDone(ctx context.Context, nextInterval time.Duration) e
 		close(sleepCh)
 	}()
 	select {
+	case <-timeouter.C:
+		return fmt.Errorf("retrier timed out after %v", r.timeout)
```

Comment:
possibly this should be a public error value so that people can test for `errors.Is(err, roko.TimeoutExceeded)` or something